lib/Sema/PreCheckTarget.cpp

```diff
@@ -331,6 +331,10 @@ static Expr *getMemberChainSubExpr(Expr *expr) {
     return FVE->getSubExpr();
   } else if (auto *SE = dyn_cast<SubscriptExpr>(expr)) {
     return SE->getBase();
+  } else if (auto *DSE = dyn_cast<DotSelfExpr>(expr)) {
+    return DSE->getSubExpr();
+  } else if (auto *PUE = dyn_cast<PostfixUnaryExpr>(expr)) {
+    return PUE->getOperand();
```

Comment:
I don't think we want to apply this to unresolved member chains, e.g for:

```swift
postfix operator ^
postfix func ^ (_ lhs: S) -> Int { 0 }

struct S {
  static var x = S()
}
func foo(_ x: Int) {}
foo(.x^)
```

The member chain should be `.x`, not `.x^`, as otherwise that would fail to type-check since the inference would be coming from `foo` instead of `^`. Seems like this is something that should be specific to optional chains.

Reply:
I wonder if this could be handled in a similar way to AssignExpr?

Reply:
Good point! Fixed and added an implicit member expression test case.